activemodel/lib/active_model/attribute.rb

```diff
@@ -124,15 +127,15 @@ def hash
       [self.class, name, value_before_type_cast, type].hash
     end
 
-    def init_with(coder)
+    def init_with(coder = EMPTY_HASH)
       @name = coder["name"]
       @value_before_type_cast = coder["value_before_type_cast"]
       @type = coder["type"]
       @original_attribute = coder["original_attribute"]
-      @value = coder["value"] if coder.map.key?("value")
```

Comment:
This mapper can't be an enumerable. It is always an hash. See https://github.com/ruby/psych/blob/master/lib/psych/coder.rb#L14